lib/cisco_node_utils/vxlan_vtep_vni.rb

```diff
@@ -24,13 +24,19 @@
 module Cisco
   # VxlanVtepVni - node utility for vxlan vtep vni members.
   class VxlanVtepVni < NodeUtil
-    attr_reader :name, :vni
+    attr_reader :name, :vni, :vrf
 
-    def initialize(name, vni, instantiate=true)
+    def initialize(name, vni, vrf=false, instantiate=true)
       @name = name
       @vni = vni
+      @vrf = vrf
 
       create if instantiate
+      # NOTE: This can be removed after the puppet type/provider are coded
+      # but a few design notes so we don't forget!.
+      # 1) The title pattern should be cisco_vxlan_vtep_vni { 'name', 'vni' :
+      # 2) vrf will be a property but will also be a namevar.
+      # 3) The type must check that vrf is always set to either true or false.
```

Comment:
We don't have any puppet code for this yet and I figured whoever works on puppet will likely work with the nodeutils code as well and very likely see this.  I was supposed to work on the puppet side of this next but if you get to it first... consider yourself warned :)